Intersect.Server/Entities/Entity.cs

```diff
@@ -2483,36 +2491,33 @@ protected bool IsOneBlockAway(Entity target)
         //These functions only work when one block away.
         protected bool IsFacingTarget(Entity target)
         {
-            if (IsOneBlockAway(target))
+            if (!IsOneBlockAway(target))
             {
-                var myTile = new TileHelper(MapId, X, Y);
-                var enemyTile = new TileHelper(target.MapId, target.X, target.Y);
-                myTile.Translate(0, -1);
-                if (myTile.Matches(enemyTile) && Dir == (int) Directions.Up)
-                {
-                    return true;
-                }
+                return false;
+            }
 
-                myTile.Translate(0, 2);
-                if (myTile.Matches(enemyTile) && Dir == (int) Directions.Down)
-                {
-                    return true;
-                }
+            var myTile = new TileHelper(MapId, X, Y);
+            var enemyTile = new TileHelper(target.MapId, target.X, target.Y);
+            myTile.Translate(0, -1);
+            if (myTile.Matches(enemyTile) && Dir == (int)Directions.Up)
+            {
+                return true;
+            }
 
-                myTile.Translate(-1, -1);
-                if (myTile.Matches(enemyTile) && Dir == (int) Directions.Left)
-                {
-                    return true;
-                }
+            myTile.Translate(0, 2);
+            if (myTile.Matches(enemyTile) && Dir == (int)Directions.Down)
+            {
+                return true;
+            }
 
-                myTile.Translate(2, 0);
-                if (myTile.Matches(enemyTile) && Dir == (int) Directions.Right)
-                {
-                    return true;
-                }
+            myTile.Translate(-1, -1);
+            if (myTile.Matches(enemyTile) && Dir == (int)Directions.Left)
+            {
+                return true;
             }
 
-            return false;
+            myTile.Translate(2, 0);
+            return myTile.Matches(enemyTile) && Dir == (int)Directions.Right;
```

Comment:
This is horrible but I am not about to write the code to fix it, just add `// TODO: Get rid of this horrible code`

Reply:
I was thinkinh about reverting this back and avoid touching it since the entire method will be changed with the diagonal movement PR anyways.

Reply:
Fair. We can revisit this PR after the diagonal PR is merged then? Mark this as a draft if you are ok with that.